### ISSUE TITLE
Missing native type hint even though it cannot be added on overloaded class method

### ISSUE THREAD

**Issue by jahudka** [slevomat/coding-standard#1218]
The `SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint` sniff suggests adding an `array` type hint to a parameter of an overloaded class method even though it's not present in the parent class either, which is wrong, because adding that type hint breaks code. Consider the following example:

```php
namespace Foo\Bar\Baz;

use Doctrine\DBAL\Schema\Column;
use Doctrine\DBAL\Schema\PostgreSqlSchemaManager;

class MySchemaManager extends PostgreSqlSchemaManager
{
  /** @param mixed[] $tableColumn */
  protected function _getPortableTableColumnDefinition($tableColumn) : Column
  {
    // ...
  }
}
```

The sniff will report the following error:

```
Method Foo\Bar\Baz\MySchemaManager::_getPortableTableColumnDefinition() does not have native type hint
for its parameter $tableColumn but it should be possible to add it based on @param annotation "mixed[]".
```

The [parent class](https://github.com/doctrine/dbal/blob/2.13.x/lib/Doctrine/DBAL/Schema/PostgreSqlSchemaManager.php#L324) does not specify an `array` type hint for the `$tableColumn` parameter of this method, therefore specifying it in the descendant class isn't allowed as it would break contravariance.

The sniff not only suggests you break your code, it will break it for you when you run `phpcbf`, because it is automatically "fixable". That seems like a pretty dangerous thing to me since people have been known to run `phpcbf` as part of a CD pipeline..

The same thing happens with the `mixed` type on PHP 8, but as `mixed` means "anything goes" and is functionally equivalent to not specifying a type hint at all, PHP doesn't seem to mind.

Inheritance in general seems to be a bit of a neglected feature where PHPCS is concerned - for example the above code would also generate an `PSR2.Methods.MethodDeclaration.Underscore` error, even though it should be obvious to a static analysis tool that the method is being overloaded and therefore removing the underscore is not an option.. but that's a whole different battle.

**Comment by kukulich** [comment]
Yes, PHPCS works in one-file scope.

**Comment by jahudka** [comment]
wait... that's _it_? 😮 like, no explanation, no link to a lengthy discussion about why the described behaviour is considered correct, just a single bare sentence and issue closed?

umm..

okay..
